sdk/cosmos/azure-cosmos-spark_3_2-12/pom.xml

```diff
@@ -51,7 +51,7 @@
     <dependency>
       <groupId>org.scala-lang</groupId>
       <artifactId>scala-library</artifactId>
-      <version>2.12.10</version> <!-- {x-version-update;cosmos_org.scala-lang:scala-library;external_dependency} -->
+      <version>2.12.15</version> <!-- {x-version-update;cosmos_org.scala-lang:scala-library;external_dependency} -->
```

Comment:
Since we mark this as provided we can't justc hange it in the common package

Two options:
1) Keep it in the older version - should be backwards compatible
2) If 3.5 requires newer version make this a Spark-version specific dependency - like put it in Spark 3.1, 3.2 etc. separately with the matching version